src/dmd/func.d

```diff
@@ -154,11 +154,40 @@ enum FUNCFLAG : uint
 
 /***********************************************************
  */
+
+extern (C++) struct Ensure
```

Comment:
Wouldn't this be better as `class EnsureStatement : Statement`? Then it can go in the statement module.

Reply:
It is not really a statement, it's just a way to create a tuple of result identifier and out contract.
I can of course declare it as a statement if that is better.